src/conn.c

```diff
@@ -21,20 +21,38 @@ struct rpma_conn {
 /* internal librpma API */
 
 /*
- * rpma_conn_new -- wrap provided entities into a newly created connection
- * object
+ * rpma_conn_new -- migrate an obtained CM ID into newly created event channel.
+ * If succeeded wrap provided entities into a newly created connection object.
+ *
+ * Note: rdma_migrate_id(3) will block if the previous event channel of the CM
```

Comment:
Is this common? That sounds like a possible deadlock to me. Can we prevent it somehow? Maybe check that there are any outstanding events and return E_WOULD_BLOCK?